tests/zentorch/test_sanity.py

```diff
@@ -0,0 +1,90 @@
+import os
+
+import pytest
+import torch
+import zentorch  # noqa: F401
+from testing_utils import (
+    FAST_DIFFUSION_KWARGS,
+    FAST_TEXT_GENERATION_KWARGS,
+    SUPPORTED_DIFFUSION_PIPELINES_TINY,
+    SUPPORTED_SIMPLE_MODELS_TINY,
+    SUPPORTED_TEXT_GENERATION_MODELS_TINY,
+    compile_diffusion_pipeline,
+    compile_transformers_model,
+    get_diffusion_pipeline_inputs,
+    get_transformers_model_inputs,
+    load_diffusion_pipeline,
+    load_transformers_model,
+)
+
+
+def test_amdrun_zentorch_setup():
+    assert os.environ["OMP_DYNAMIC"] == "False"
+    assert os.environ["ZENDNN_GEMM_ALGO"] == "4"
+    assert os.environ["OMP_WAIT_POLICY"] == "ACTIVE"
+    assert os.environ["OMP_NUM_THREADS"] == f"{os.cpu_count()}"
+    assert os.environ["GOMP_CPU_AFFINITY"] == f"0-{os.cpu_count() - 1}"
+    assert os.environ["LD_PRELOAD"] == "/usr/lib/x86_64-linux-gnu/libjemalloc.so"
+    assert (
+        os.environ["MALLOC_CONF"]
+        == "oversize_threshold:1,background_thread:true,metadata_thp:auto,dirty_decay_ms:-1,muzzy_decay_ms:-1"
+    )
+
+
+@pytest.mark.parametrize("model_type", SUPPORTED_SIMPLE_MODELS_TINY.keys())
+def test_simple_models(model_type: str):
+    model_id_and_tasks = SUPPORTED_SIMPLE_MODELS_TINY[model_type]
+
+    for model_id, tasks in model_id_and_tasks.items():
+        for task in tasks:
+            inputs = get_transformers_model_inputs(model_id, task)
+
```

Comment:
I would add a `model.eval()` & `with torch.no_grad():`

Actually, it could be helpful to have a test as well about gradients, as compile is about but the forward & backward (can be done later of course)

Reply:
just realized I removed the `inference_mode` context at a certain point. will put it back.